collect_stata/write_json.py

```diff
@@ -0,0 +1,308 @@
+    )
+    dataframe["labels"] = dataframe["labels"].astype(str)
+    dataframe["values"] = pd.to_numeric(dataframe["values"])
+    dataframe.sort_values(by="values", inplace=True)
+    return dataframe.to_dict("list")
+
+
+def uni_cat(elem, file_csv):
+    """Generate dict with frequencies and labels for categorical variables
+
+    Input:
+    elem: dict
+    file_csv: pandas DataFrame
+
+    Output:
+    cat_dict: dict
+    """
+
+    frequencies = []
+    values = []
+    missings = []
+    labels = []
+
+    value_count = file_csv[elem["name"]].value_counts()
+    for value in elem["values"]:
+        try:
+            frequencies.append(int(value_count[value["value"]]))
+        except KeyError:
+            frequencies.append(0)
+        labels.append(value["label"])
+
+        var_value = value["value"]
+
+        if int(value["value"]) >= 0:
+            missings.append(False)
+            values.append(var_value)
+        else:
+            missings.append(True)
+            values.append(var_value)
+
+    cat_dict = sorting_dataframe(values, labels, missings, frequencies)
+
+    return cat_dict
+
+
+def uni_string():
+    """Generate dict with frequencies for nominal variables
+
+    Output:
+    string_dict: OrderedDict
+    """
+
+    string_dict = OrderedDict()
+
+    string_dict["frequencies"] = []
+    string_dict["labels"] = []
+    string_dict["missings"] = []
+    string_dict["values"] = []
+
+    return string_dict
+
+
+def uni_number():
+    """Generate dict with frequencies for numerical variables
+
+    Output:
+    OrderedDict
+    """
+
+    return OrderedDict(frequencies=[], labels=[], labels_de=[], missings=[], values=[])
+
+
+def stats_cat(elem, file_csv):
+    """Generate dict with statistics for categorical variables
+
+    Input:
+    elem: dict
+    file_csv: pandas DataFrame
+
+    Output:
+    dict
+    """
+
+    total = int(file_csv[elem["name"]].size)
+    invalid = int(file_csv[elem["name"]].isnull().sum()) + int(
+        sum(n < 0 for n in file_csv[elem["name"]])
+    )
+    valid = total - invalid
+
+    return {"valid": valid, "invalid": invalid}
+
+
+def stats_string(elem, file_csv):
+    """Generate dict with statistics for nominal variables
+
+    Input:
+    elem: dict
+    file_csv: pandas DataFrame
+
+    Output:
+    dict
+    """
+
+    valid = int(file_csv[elem["name"]].value_counts().sum())
+    invalid = int(file_csv[elem["name"]].isnull().sum())
+    for i in file_csv[elem["name"]]:
+        if i in ("", "."):
+            valid = valid - 1
+            invalid = invalid + 1
+
+    return {"valid": valid, "invalid": invalid}
+
+
+def stats_number(elem, file_csv):
+    """Generate dict with statistics for numerical variables
+
+    Input:
+    elem: dict
+    file_csv: pandas DataFrame
+
+    Output:
+    statistics: OrderedDict
+    """
+
+    data_withoutmissings = file_csv[file_csv[elem["name"]] >= 0][elem["name"]]
+
+    first_q, third_q = np.quantile(data_withoutmissings, [0.25, 0.75])
```

Comment:
This is better than before.

But you operate over every list / array a couple of times, quantile, mean, median, min, max. They all need to traverse the array and or sort I guess?

How about this @hansendx?
```python
summary = data_withoutmissings.describe()
# prepare total, valid, invalid
return {
    "Min.": summary["min"],
    "1st Qu.": summary["25%"],
    "Median": summary["50%"],
    "Mean": summary["mean"],
    "3rd Qu.": summary["75%"],
    "Max.": summary["max"],
    "valid": valid,
    "invalid": invalid,
}
```
I think pandas' describe() is doing the calcuations in one swing?